source/scripts/edit.js

```diff
@@ -1,18 +1,21 @@
+import { getTasksFromStorage, saveTaskToStorage } from "./add.js";
 // --------------------------edit popup window--------------------------------------
 
 /* add functions to the edit button on-click */
-document.getElementById("editButton").addEventListener("click", () => {openEditForm(0);});
+// document.getElementById("editButton").addEventListener("click", () => {openEditForm(0);});
 document.getElementById("cancelEditButton").addEventListener("click", closeEditForm);
 
 /**
  * Open Edit button function
  * Once the user clicks on the add button, the edit popup form should pops up
  * The ID of the current task is also passed onto the deleteEditButton to reference the task upon deletion
+ * as well as upon saving new data
  * @param {number} id ID of the task to be edited
  */
 function openEditForm(id) {
     document.getElementById("editForm").style.display = "block";
     document.getElementsByClassName("deleteEditButton")[0].id = id;
+    document.getElementsByClassName("submitEditButton")[0].id = id;
```

Comment:
We can avoid setting the same `id` of the `deleteEditButton` and `submitEditButton` by just not giving `submitEditButton` an `id`. On line 34 of `edit.js`, instead of calling `saveData(document.getElementsByClassName("submitEditButton")[0].id);` , we can call `saveData(document.getElementsByClassName("deleteEditButton")[0].id);`. This line of code saves the updated data into the respective `task id`. It still functions correctly since we are simply using the `id` of the `deleteEditButton` as a reference as to which `task id` it should save the data to. Thus eliminating assigning the same `id` twice.

Reply:
Refactored code as explained in [a9d0dc2](https://github.com/cse110-fa22-group21/cse110-fa22-group21/pull/68/commits/a9d0dc2822b4af8ed9d63d4d80c275b665f2236a).